Test/AwsCommonRuntimeKitTests/mqtt/Mqtt5ClientTests.swift

```diff
@@ -114,4 +252,666 @@ class Mqtt5ClientTests: XCBaseTestCase {
+        }
+
+        try client.stop()
+        if testContext.semaphoreDisconnection.wait(timeout: .now() + 5) == .timedOut {
+            print("Disconnection timed out after 5 seconds")
+            XCTFail("Disconnection timed out")
+        }
+
+        if testContext.semaphoreStopped.wait(timeout: .now() + 5) == .timedOut {
+            print("Stop timed out after 5 seconds")
+            XCTFail("Stop timed out")
+        }
+    }
+
+    /*===============================================================
+                     WEBSOCKET CONNECT TEST CASES
+    =================================================================*/
+    // TODO implement websocket tests after websockets are implemented
+
+    /*===============================================================
+                     NEGATIVE CONNECT TEST CASES
+    =================================================================*/
+
+    /*
+     * [ConnNegativeID-UC1] Client connect with invalid host name
+     */
+    func testMqtt5DirectConnectWithInvalidHost() throws {
+
+        let clientOptions = MqttClientOptions(
+            hostName: "badhost",
+            port: UInt32(1883))
+
+        let testContext = MqttTestContext()
+
+        let client = try createClient(clientOptions: clientOptions, testContext: testContext)
+        try client.start()
+
+        if testContext.semaphoreConnectionFailure.wait(timeout: .now() + 5) == .timedOut {
+            print("Connection Failure Timed out after 5 seconds")
+            XCTFail("Connection Timed Out")
+        }
+
+        if let failureData = testContext.lifecycleConnectionFailureData {
+            XCTAssertEqual(failureData.crtError.code, Int32(AWS_IO_DNS_INVALID_NAME.rawValue))
+        } else {
+            XCTFail("lifecycleConnectionFailureData Missing")
+        }
+
+        try client.stop()
+
+        if testContext.semaphoreStopped.wait(timeout: .now() + 5) == .timedOut {
+            print("Stop timed out after 5 seconds")
+            XCTFail("Stop timed out")
+        }
+    }
+
+    /*
+     * [ConnNegativeID-UC2] Client connect with invalid port for direct connection
+     */
+    func testMqtt5DirectConnectWithInvalidPort() throws {
+        let inputHost = try getEnvironmentVarOrSkipTest(environmentVarName: "AWS_TEST_MQTT5_DIRECT_MQTT_HOST")
+
+        let clientOptions = MqttClientOptions(
+            hostName: inputHost,
+            port: UInt32(444))
+
+        let testContext = MqttTestContext()
+
+        let client = try createClient(clientOptions: clientOptions, testContext: testContext)
+        try client.start()
+
+        if testContext.semaphoreConnectionFailure.wait(timeout: .now() + 5) == .timedOut {
+            print("Connection Failure Timed out after 5 seconds")
+            XCTFail("Connection Timed Out")
+        }
+
+        if let failureData = testContext.lifecycleConnectionFailureData {
+            if failureData.crtError.code != Int32(AWS_IO_SOCKET_CONNECTION_REFUSED.rawValue) &&
```

Comment:
Checkout the test doc, the error seems to be the following: 
```
        * error_code == AWS_ERROR_MQTT5_DECODE_PROTOCOL_ERROR (444)
        * error_code == AWS_ERROR_SYS_CALL_FAILURE (direct connect port)
```
However, when I tested it locally, I also get `AWS_IO_SOCKET_CONNECTION_REFUSED` instead. Was the doc outdated?